lib/ansible/modules/blockinfile.py

```diff
@@ -287,11 +298,11 @@ def main():
 
     insertbefore = params['insertbefore']
     insertafter = params['insertafter']
-    block = to_bytes(params['block'])
-    marker = to_bytes(params['marker'])
+    block = params['block']
+    marker = params['marker']
     present = params['state'] == 'present'
-    b_linesep = os.linesep.encode()
-    blank_line = [b_linesep]
+
+    blank_line = [os.linesep]
```

Comment:
```suggestion
    line_separator = os.linesep
    blank_line = [line_separator]
```
By using a single variable, we only need to change the value in one place, reducing the chance of errors and making the code more maintainable. not a blocker but a suggestion.